lib/matplotlib/path.py

```diff
@@ -667,14 +667,34 @@ def intersects_bbox(self, bbox, filled=True):
 
     def interpolated(self, steps):
         """
-        Return a new path resampled to length N x *steps*.
+        Return a new path with each segment divided into *steps* parts.
 
-        Codes other than `LINETO` are not handled correctly.
+        Codes other than `LINETO` `MOVETO` and `CLOSEPOLY` are not handled correctly.
+
+        Parameters
+        ----------
+        steps : int
+            The number of segments in the new path for each in the original.
+
+        Returns
+        -------
+        Path
+            The interpolated path.
         """
         if steps == 1:
             return self
 
-        vertices = simple_linear_interpolation(self.vertices, steps)
+        if self.codes is not None and self.MOVETO in self.codes[1:]:
+            return self.make_compound_path(
+                *(p.interpolated(steps) for p in self._iter_connected_components()))
+
+        if self.codes[-1] == self.CLOSEPOLY and not np.all(self.vertices[-1] ==
+                                                           self.vertices[0]):
```

Comment:
I am assuming that after splitting by `MOVETO`, any `CLOSEPOLY` will be at the end of the path.  Is that correct?

Reply:
It looks like `_iter_connected_components` only splits on `MOVETO`, so it wouldn't notice `CLOSEPOLY`. I'm not sure what happens to a `Path` with `CLOSEPOLY` stuck in the middle.

Reply:
A CLOSEPOLY the middle can only happen if it was not followed by a MOVETO in the original path. But that means you closed a shape, and continue to draw from that position. People would likely just use LINETOs in such a case, But AFAIK there's nothing in the rules prohibiting `LINETO, CLOSEPOLY, LINETO`, and there could be rare cases where it makes sense/could happen.

That said, I haven't checked what we actually do here and whether that pattern renders reasonably.

Reply:
With latest update

![Figure_1](https://github.com/user-attachments/assets/b7398e5d-fb3a-4e60-9595-be0de084e241)

Reply:
> Edit: given the axes limits here go down beyond (0,0), I guess that whatever tells `autoscale` the extent of the path also does not know to ignore `CLOSEPOLY`.

I'd say this is a separate topic. Without checking: I suspect that autoscale just takes all the points, no matter what the associated code is (likely also the control points for curve).

At least for `CLOSEPOLY` people can work around this by using existing coordinates.

Reply:
> I'd say this is a separate topic.

Sure, that was very much an aside - not proposing to investigate it here.